src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -254,7 +253,8 @@ private TypeSymbol GetTypeSymbol()
             }
             else
             {
-                declType = typeBinder.BindType(_typeSyntax, diagnostics, out isVar);
+                RefKind refKind;
+                declType = typeBinder.BindType(_typeSyntax.SkipRef(out refKind), diagnostics, out isVar);
```

Comment:
>declType [](start = 16, length = 8)

Actually, we should probably not skip ref here (we want to give a diagnostic if ref is contained in the type at this point).